syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message PoseDetectFloatsCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional PoseDetectFloatsCalculatorOptions ext = 236399889;
  }
  enum InputSize {
    UNKNOWN = 0;
    INPUT_1D = 1;
    INPUT_2D = 2;
  }

  // If input_size is INPUT_2D, unpack a vector<vector<float>> to a
  // 2d tensor (matrix). If INPUT_1D,
  // convert a vector<float> into a 1d tensor (vector).
  optional InputSize input_size = 1 [default = INPUT_1D];

  // If true, the output tensor is transposed.
  // Otherwise, the output tensor is not transposed.
  // It will be ignored if tensor_is_2d is INPUT_1D.
  optional bool transpose = 2 [default = false];
}
